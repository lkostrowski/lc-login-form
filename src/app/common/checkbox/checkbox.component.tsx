import React, { HTMLProps } from 'react';
import cx from 'classnames';

import checkboxNotFilledUrl from '../../../icons/checkbox-unchecked.svg';
import checkboxFilledUrl from '../../../icons/checkbox.svg';

import styles from './checkbox.module.scss';

interface Props extends HTMLProps<HTMLInputElement> {
    label: string;
}

export const Checkbox: React.FC<Props> = ({
    className,
    checked,
    label,
    ...props
}) => {
    return (
        <label className={cx(className, styles.container)}>
            <input checked={checked} hidden type="checkbox" {...props} />
            {checked ? (
                <img className={styles.icon} alt="" src={checkboxFilledUrl} />
            ) : (
                <img
                    className={styles.icon}
                    alt=""
                    src={checkboxNotFilledUrl}
                />
            )}
            <span className={styles.label}>{label}</span>
        </label>
    );
};
