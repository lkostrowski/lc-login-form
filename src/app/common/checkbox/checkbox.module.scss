.container {
    display: flex;
    align-items: center;
    cursor: pointer;
}

.icon {
    margin-right: 1rem;
}

.label {
    user-select: none;
}
