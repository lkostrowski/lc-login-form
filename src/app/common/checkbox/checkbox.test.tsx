import { cleanup, render } from '@testing-library/react';
import React from 'react';

import { Checkbox } from './checkbox.component';

afterEach(cleanup);

describe('Checkbox component', () => {
    describe('Snapshots', () => {
        test('Basic version', () => {
            const { container } = render(<Checkbox label='Checkbox' />);

            expect(container).toMatchSnapshot();
        });
        test('Checked version', () => {
            const { container } = render(
                <Checkbox onChange={() => {}} checked label='Checkbox' />,
            );

            expect(container).toMatchSnapshot();
        });
    });
});
