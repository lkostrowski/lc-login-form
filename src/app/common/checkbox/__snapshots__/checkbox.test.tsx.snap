// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component Snapshots Basic version 1`] = `
<div>
  <label
    class="container"
  >
    <input
      hidden=""
      type="checkbox"
    />
    <img
      alt=""
      class="icon"
      src="checkbox-unchecked.svg"
    />
    <span
      class="label"
    >
      Checkbox
    </span>
  </label>
</div>
`;

exports[`Checkbox component Snapshots Checked version 1`] = `
<div>
  <label
    class="container"
  >
    <input
      checked=""
      hidden=""
      type="checkbox"
    />
    <img
      alt=""
      class="icon"
      src="checkbox.svg"
    />
    <span
      class="label"
    >
      Checkbox
    </span>
  </label>
</div>
`;
